### PR TITLE
Make the ReadOnlyArrayMap extend AbstractMap for equals/hashcode implementation

### PR DESCRIPTION
**Pull request by jkwatson**
This makes our map implementation do proper equals/hashcode with other maps that contain the same key/value pairs.

### CODE REVIEW

File: api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class ReadOnlyArrayMapTest {
+
+  @Test
+  void equalsHashCode() {
```
Comment (bogdandrutu) [inline api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java:17]: FYI: Guava has a nice testing util for equal and hash.

Comment (jkwatson) [reply api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java:18]: I didn't see anything for hashcode, but only the EqualsTester. Did I miss something for hashcode?

Comment (bogdandrutu) [reply api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java:18]: EqualTester tests for both and all possible combinations that you give see https://www.javadoc.io/doc/com.google.guava/guava-testlib/21.0/com/google/common/testing/EqualsTester.html

**Review by bogdandrutu — APPROVED** [review]

File: api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    Map<String, String> one = ReadOnlyArrayMap.wrap(Arrays.asList("a", "b"));
+    Map<String, String> two = ReadOnlyArrayMap.wrap(Arrays.asList("a", "b"));
+    Map<String, String> three = ReadOnlyArrayMap.wrap(Arrays.asList("c", "d"));
+
+    new EqualsTester().addEqualityGroup(one, two).addEqualityGroup(three).testEquals();
```
Comment (bogdandrutu) [inline api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java:21]: Consider to add an empty list as well in one of the groups

**Review by bogdandrutu — APPROVED** [review]

File: api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java (hunk 2 above)
Comment (jkwatson) [reply api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java]: done

**Pull request merged**